plugins/filter/filters.py

```diff
@@ -228,23 +228,38 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
                     oauth_enabled and rbac_enabled:
-                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
-                    oauth_pem_path + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'io.confluent.kafka.server.plugins.auth.token.CompositeBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' \
                     and oauth_enabled and not rbac_enabled:
-                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
+                    oauth_enabled:
+                if rbac_enabled and idp_self_signed:
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
```

Comment:
Also I dont understand the reason why publicKeyPath is only present in case of rbac_enabled is true

Reply:
That is the mds key path, there is no mds in non rbac

Reply:
Oh in that case should we rename it to something like rbac_enabled_public_pem_path so it doesnt confuse as something which is oauth specific

Reply:
sorry even I got confused with the previous comment, it should not be the same as oauth cert path, it is the mds pem file. I understand the names are confusing, should be changed

Reply:
updated the name to rbac_enabled_public_pem_path